src/main/java/bio/terra/service/duos/DuosDataset.java

```diff
@@ -0,0 +1,8 @@
+package bio.terra.service.duos;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.google.common.annotations.VisibleForTesting;
+
+@VisibleForTesting
+@JsonIgnoreProperties(ignoreUnknown = true)
+public record DuosDataset(Integer dataSetId) {}
```

Comment:
Is `dataSetId` nullable in practice? If not, would use `int` here work instead? That would make it clearer that `null` isn't allowed here.
```suggestion
public record DuosDataset(int dataSetId) {}
```

Reply:
This is a good question!  While in practice we don't expect this to be null I wanted to match this exactly to DUOS' defined model in their code base, since we don't have a client from them:

https://github.com/DataBiosphere/consent/blob/6ec50f6b9ddc1649e17b1f0522107eea3084dddb/src/main/java/org/broadinstitute/consent/http/models/Dataset.java#L12

Reply:
I will also say: given that we don't really control it, I always err on the side of boxed objects (I'm allergic to NPEs :) )

Reply:
Using boxed types doesn't avoid NPEs, it only defers them.

The problem with boxed types is you can end up with NPEs without knowing it, as java won't flag a compile error if you try to use an `Integer` in a math expression, and you still end up with an NPE when the boxed object is `null`. So using boxed types only avoids NPEs if you never use the boxed type as a non-boxed type, or are very careful to check for nulls before using them.

But in this case it's OK since it's a copied type from another API and the code in this PR never looks at the content of this object. If we do ever add code that uses this field, we'll have to be careful.

Reply:
IMO it's easier to debug with boxed types.  E.g. if I try to access it, it tells me "boom, I'm null" and points me to the actual problem area.  Not so with the unboxed type.  You end up getting weird stack traces

Reply:
It's hard to disagree with wanting to avoid weird stack traces...

The benefit I see to not using boxed types is that the compiler will flag an error if you try to assign `null` to an object that can't be null. The IDE can also warn in a case where a method that it knows can return `null` is used. So this turns what would be a runtime error into a compile time error.

But this only works if a field can never be `null`. If it can possibly be null, then you have to use a boxed type. If it can't possibly be null, then used an unboxed type. If it's somewhere in between -- it should never be `null`, but we can't rule it out completely -- then you have to decide which way you want to find out about the error.